video2dataset/downloader.py

```diff
@@ -24,19 +25,31 @@ def handle_youtube(youtube_url):
     return cv2_vid, dst_name
 
 
-def handle_mp4_link(mp4_link):
-    resp = requests.get(mp4_link, stream=True)
-    ntf = tempfile.NamedTemporaryFile()  # pylint: disable=consider-using-with
-    ntf.write(resp.content)
-    ntf.seek(0)
-    dst_name = mp4_link.split("/")[-1][:-4] + ".npy"
-    return ntf, dst_name
+def handle_mp4_link(
+    mp4_link: str,
+    extract_audio: bool = False,
+    sample_rate: int = None
+):
+    if extract_audio:
+        audio_stream, audio_info = get_info_and_resample(
+            mp4_link, sample_rate)  # returns bytes of resampled audio and audio info
+        return None, audio_stream, audio_info, None
+
+    else:
+        resp = requests.get(mp4_link, stream=True)
+        ntf = tempfile.NamedTemporaryFile()  # pylint: disable=consider-using-with
+        ntf.write(resp.content)
+        ntf.seek(0)
+        dst_name = mp4_link.split("/")[-1][:-4] + ".npy"
+        return ntf.name, ntf, None, dst_name
 
 
-def handle_url(url):
+def handle_url(url, sample_rate=None, get_audio=False):
```

Comment:
both parameters not necessary. If sample_rate is not None then get_audio must be True, just use that

Reply:
I removed get_audio argument